MapboxVision/src/main/java/com/mapbox/vision/sync/SessionManager.kt

```diff
@@ -0,0 +1,268 @@
+            if (isRecording) {
+                VisionLogger.e(TAG_CLASS, "Recording was already started.")
+                return
+            }
+            isRecording = true
+
+            sessionWriter.stop()
+            sessionWriter = createRecordingSessionWriter(path)
+            sessionWriter.start()
+
+            telemetrySyncManager.stop()
+            videoProcessor.detach()
+        }
+
+        override fun stopRecording() {
+            if (!isRecording) {
+                VisionLogger.e(TAG_CLASS, "Recording was not started.")
+                return
+            }
+            isRecording = false
+
+            sessionWriter.stop()
+            sessionWriter = createRotatedBuffersSessionWriter()
+            sessionWriter.start()
+
+            videoProcessor.attach(videoProcessorListener)
+            telemetrySyncManager.start()
+            checkCountryTelemetryDir()
+        }
+
+        private fun checkCountryTelemetryDir() {
+            val telemetryDir = getCountryTelemetryDir()
+
+            telemetryDir?.let { dir ->
+                File(dir).listFiles()?.forEach {
+                    if (it.list().isNullOrEmpty()) {
+                        it?.delete()
+                    } else {
+                        syncSessionDir(it.absolutePath)
+                    }
+                }
+            }
+        }
+
+        private fun createRotatedBuffersSessionWriter() = RotatedBuffersSessionWriter(
+            buffers = RotatedBuffers(getVideoBuffersDir()),
+            nativeVisionManager = nativeVisionManager,
+            rootCacheDir = getCacheDir(),
+            videoRecorder = videoRecorder,
+            telemetryImageSaverImpl = telemetryImageSaver as TelemetryImageSaverImpl,
+            sessionWriterListener = sessionWriterListener
+        )
+
+        private fun createRecordingSessionWriter(path: String) = RecordingSessionWriter(
+            buffers = RotatedBuffers(buffersDir = "$path/", totalBuffersNumber = 1),
+            nativeVisionManager = nativeVisionManager,
+            sessionDir = "$path/",
+            videoRecorder = videoRecorder
+        )
+
+        private fun configMapboxTelemetry() {
+            isBaseUrlSet = try {
+                // TODO remove when fix is no more necessary
+                mapboxTelemetry.updateDebugLoggingEnabled(BuildConfig.DEBUG)
+                mapboxTelemetry.setBaseUrl(telemetryEnvironment.getHost(currentCountry))
+                true
+            } catch (e: Exception) {
+                false
+            }
+        }
+
+        private val sessionWriterListener = object : SessionWriterListener {
+            override fun onSessionStop(
+                clips: Array<VideoClip>,
+                videoPath: String,
+                outputPath: String,
+                sessionStartMillis: Long
+            ) {
+                if (currentCountry != Country.Unknown) {
+
+                    val sessionCountryDir = getCountryTelemetryDir()
```

Comment:
```suggestion
                    val sessionCountryDir = getCountryTelemetryDir() ?: return
```
and remove `.let{ }` block

Reply:
done